numba/tests/npyufunc/test_gufunc.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import absolute_import, print_function, division
+from numba import unittest_support as unittest
+from numba.decorators import jit
+from numba import float32
+import numpy as np
+import numpy.core.umath_tests as ut
+from numba.npyufunc import GUVectorize
+
+
+def matmulcore(A, B, C):
+    m, n = A.shape
+    n, p = B.shape
+    for i in range(m):
+        for j in range(p):
+            C[i, j] = 0
+            for k in range(n):
+                C[i, j] += A[i, k] * B[k, j]
+
+
+class TestGUFunc(unittest.TestCase):
+    def test_numba(self):
+        jit_matmulcore = jit((float32[:, :], float32[:, :], float32[:,:]))(matmulcore)
+
+        A = np.arange(16, dtype=np.float32).reshape(4, 4)
+        B = np.arange(16, dtype=np.float32).reshape(4, 4)
+        C = np.zeros(16, dtype=np.float32).reshape(4, 4)
+        Gold = np.matrix(A) * np.matrix(B)
+
+        jit_matmulcore(A, B, C)
+
+        self.assertTrue((C == Gold).all())
+
+    def test_gufunc(self):
+        _test_gufunc(self, 'cpu')
+
+
+def _test_gufunc(self, target):
```

Comment:
I was wondering the same. I have now manually "inlined" the _test_gufunc() function. I don't know why it was separate in the first place.